### DISCUSSION TITLE
Storybook a11y addon not failing on custom object-alt rule

### DISCUSSION THREAD

**Discussion by armper** [storybookjs/storybook#28945] [Help]
### Summary

I'm using Storybook with the @storybook/addon-a11y addon to ensure my components meet accessibility standards. I've enabled the object-alt rule and added a custom check for it, but it doesn't seem to fail even when the custom check returns false.
Here is my preview.ts file:

```import type { Preview } from "@storybook/angular";
import { setCompodocJson } from "@storybook/addon-docs/angular";
import docJson from "../documentation.json";
import { moduleMetadata } from '@storybook/angular';
import { IconModule } from 'carbon-components-angular/icon';
import { ButtonModule, TabsModule } from "carbon-components-angular";

setCompodocJson(docJson);

// Function to add the "white" theme class to the body
const applyTheme = () => {
  const document: Document = window.document;
  document.body.classList.add('white');
};

applyTheme();

const preview: Preview = {
  parameters: {
    docs: {
      toc: true, // 👈 Enables the table of contents
    },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
    a11y: {
      config: {
        rules: [
          {
            id: 'wcag2aa',
            enabled: true,
          },
          {
            id: 'object-alt',
            enabled: true, // Ensure the object-alt rule is enabled
          },
        ],
        checks: [
          {
            id: 'object-alt',
            evaluate: function (node, options) {
              console.log('Evaluating object-alt for node:', node);
              // Custom evaluation logic for object-alt
              return false; // Simulate a failure
            },
          },
        ],
      },
    },
  },
  tags: ['autodocs'], // 👈 Enables auto-generated documentation for all stories
};

export const decorators = [
  moduleMetadata({
    imports: [IconModule, TabsModule, ButtonModule],
  }),
];

export default preview;
```

**Problem:**

Despite the evaluate function returning false, the object-alt rule does not show as failed in my Storybook page. I've added console.log statements to verify that the evaluate function is being called, and it is.

[![enter image description here][1]][1]

**Steps Taken:**

Verified that the object-alt rule is enabled.
Added custom evaluation logic that returns false.
3. Checked the browser console to ensure the evaluate function is being executed.

**Question:**

Why is the object-alt rule not showing as failed in my Storybook page, and how can I ensure it correctly fails when the custom check returns false?
Any help or insights would be greatly appreciated!

  [1]: https://i.sstatic.net/KgpCROGy.png

**Comment by Marklb** [comment]
I took a quick look, so I am not positive, but I think this may be a bug. Unless I just overlooked something, since I just quickly tried to throw a reproduction together, it looks like the functions are getting removed before being passed into axe. I can't remember if parameters are one of the objects that get serialized, but if so then that is probably breaking the config. My quickly thrown together code did cause the accessibility panel to display something from my config, but it had no label and just said there was an error.

I am busy with something at work, but if I get some time then I will try to confirm what is happening. At the moment I am thinking the channels serializer may be stripping the object functions, if someone wants to take a look.
